packages/cli/src/scaling/processor.ts

```diff
@@ -0,0 +1,188 @@
+import { Service } from 'typedi';
+import { BINARY_ENCODING, ApplicationError, Workflow } from 'n8n-workflow';
+import { WorkflowExecute } from 'n8n-core';
+import { Logger } from '@/Logger';
+import config from '@/config';
+import { ExecutionRepository } from '@/databases/repositories/execution.repository';
+import { WorkflowRepository } from '@/databases/repositories/workflow.repository';
+import * as WorkflowExecuteAdditionalData from '@/WorkflowExecuteAdditionalData';
+import { NodeTypes } from '@/NodeTypes';
+import type { ExecutionStatus, IExecuteResponsePromiseData, IRun } from 'n8n-workflow';
+import type {
+	Job,
+	JobId,
+	JobResult,
+	RunningJob,
+	RunningJobSummary,
+	WebhookResponseReport,
+} from './types';
+import type PCancelable from 'p-cancelable';
+
+@Service()
+export class Processor {
+	private readonly runningJobs: { [jobId: JobId]: RunningJob } = {};
```

Comment:
With `noUncheckedIndexedAccess` this would be true of any object we're using. Since `runningJobs` is private and there's only a single access already using `?.` and the key is a string, I'd prefer to keep it like this for now.

Reply:
Fair point. In that case the type could be `Record<JobId, Running>`, which is clearer than the index signature one